### DISCUSSION TITLE
Best way to wrap a struct with a `const char*` field

### DISCUSSION THREAD

**Discussion by huidong-chen** [pybind/pybind11#4556] [Q&A]
Hello,

I am creating python binding for a C++ struct that I cannot modify, as below:

```C++
struct Information
{
    const char* name;

    // other fields
}
```

Naturally we need to create its instance as below in python:

```Python
info = Information("test", ...)
```

Therefore, I choose to implement the binding code as below:

```C++
    py::class_<Information>(m, "Information")
        .def(
            py::init(
            [](const char* name)
            {
                // deep copy is required.
                size_t const size = strlen(name) + 1;
                char* buffer = new char[size];
                memcpy(buffer, name, size);

                return new Information{ buffer };
            }))
        .def_readonly("name", &Information::name);
```

This will have memory leak because the memory allocation doesn't have a chance to be released. What is the best way to create python binding for this struct please?

Thanks in advance.

**Comment by MahmoudHussienMohamed** [comment]
Why not just using `std::string` to get rid of memory management and manual casting?
